flutter-idea/src/io/flutter/editor/FlutterIconLineMarkerProvider.java

```diff
@@ -382,18 +382,7 @@ public double findPattern(@NotNull String t, @NotNull String p) {
     return jw.similarity(t, p);
   }
 
-  static class IconInfo {
-    final @NotNull String iconName;
-    final @NotNull String className;
-    final @Nullable String familyName;
-    final @NotNull String codepoint;
-
-    IconInfo(@NotNull String className, @NotNull String iconName, @Nullable String familyName, @NotNull String codepoint) {
-      this.className = className;
-      this.iconName = iconName;
-      this.familyName = familyName;
-      this.codepoint = codepoint;
-    }
+  record IconInfo(@NotNull String className, @NotNull String iconName, @Nullable String familyName, @NotNull String codepoint) {
   }
```

Comment:
Dart has one here. I really miss the colon shorthand for empty constructor bodies.

```java
record IconInfo(@NotNull String className, @NotNull String iconName, @Nullable String familyName, @NotNull String codepoint);
```
looks nicer to my eyes.